src/miniscript/mod.rs

```diff
@@ -26,114 +26,124 @@
 
 #[cfg(feature = "serde")]
 use serde::{de, ser};
+use std::marker::PhantomData;
 use std::{fmt, str};
 
 use bitcoin;
 use bitcoin::blockdata::script;
 
 pub mod astelem;
+pub mod context;
 pub mod decode;
 pub mod lex;
 pub mod satisfy;
 pub mod types;
 
 use self::lex::{lex, TokenIter};
 use self::types::Property;
+use miniscript::decode::Terminal;
 use miniscript::types::extra_props::ExtData;
 use miniscript::types::Type;
 use std::cmp;
 use std::sync::Arc;
 use MiniscriptKey;
 use {expression, Error, ToPublicKey};
 
+pub use self::context::{Any, Legacy, ScriptContext, Segwitv0};
```

Comment:
I think we shouldn't export `Any`, it's a bit too dangerous to expose outside the library.